provisioning/service/src/Config/BulkEnrollmentOperationError.cs

```diff
@@ -0,0 +1,111 @@
+            /// A valid registration Id shall be alphanumeric, lowercase, and may contain hyphens. Max characters 128.
+            /// </remarks>
+            /// <exception cref="ArgumentException">if the provided string do not fits the registration Id requirements</exception>
+            private set
+            {
+                ParserUtils.EnsureRegistrationId(value);
+                _RegistrationId = value;
+            }
+        }
+        private string _RegistrationId;
+
+        /// <summary>
+        /// Error code.
+        /// </summary>
+        /// <remarks>
+        /// Report any error during the operation for the specific registrationId.
+        /// </remarks>
+        [JsonProperty(PropertyName = "errorCode", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public int? ErrorCode { get; private set; }
+
+        /// <summary>
+        /// Error status.
+        /// </summary>
+        /// <remarks>
+        /// Describe any error during the operation for the specific registrationId.
+        /// </remarks>
+        [JsonProperty(PropertyName = "errorStatus", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string ErrorStatus { get; private set; }
+
+        /// <summary>
+        /// Creates a new instance of <code>BulkEnrollmentOperationError</code> using information in a JSON.
+        /// </summary>
+        /// <example>
+        /// The following JSON is an example of a single error operation from a Bulk operation
+        /// <code>
+        /// {
+        ///      "registrationId":"validRegistrationId1",
+        ///      "errorCode":400,
+        ///      "errorStatus":"Bad message format exception."
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="registrationId">the <code>string</code> with a unique Id for the individualEnrollment. It cannot be <code>null</code> or empty.</param>
+        /// <param name="errorCode">the reported error code.</param>
+        /// <param name="errorStatus">the description of the error.</param>
+        /// <exception cref="ProvisioningServiceClientException">if the registrationId is null, empty or invalid</exception>
+        [JsonConstructor]
+        internal BulkEnrollmentOperationError(string registrationId, int? errorCode, string errorStatus)
+        {
+            /* SRS_BULK_ENROLLMENT_OPERATION_ERRO_21_001: [The constructor shall throws ProvisioningServiceClientException if the 
+                                                provided registrationId is null, empty, or invalid.] */
+            /* SRS_BULK_ENROLLMENT_OPERATION_ERRO_21_002: [The constructor shall store the provided information.] */
+            try
+            {
+                RegistrationId = registrationId;
+                ErrorCode = errorCode;
+                ErrorStatus = errorStatus;
+            }
+            catch (ArgumentException e)
+            {
+                throw new ProvisioningServiceClientException(e);
```

Comment:
Parser should throw the ArgumentException. If this was caused during transport (parsing data from the service), then the transport should convert to the ProvisioningServiceClientException.

Reply:
You are right, I'll change it.

Reply:
We need to propagate to the other parsers.

Reply:
Done for the ones in this PR